src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,59 @@
+package org.opentripplanner.middleware.utils;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.Leg;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public class ItineraryUtilsTest {
+    @ParameterizedTest
+    @MethodSource("createItineraryCanBeMonitoredCases")
+    public void testItineraryCanBeMonitored(Itinerary itinerary, boolean expectedResult, String message) {
+        Assertions.assertEquals(expectedResult, ItineraryUtils.itineraryCanBeMonitored(itinerary), message);
+    }
+
+    private static Stream<Arguments> createItineraryCanBeMonitoredCases() {
+        Leg transitLeg = new Leg();
+        transitLeg.mode = "BUS";
+        transitLeg.transitLeg = true;
+
+        Leg rentalLeg = new Leg();
+        rentalLeg.mode = "BICYCLE_RENT";
+        rentalLeg.rentedVehicle = true;
```

Comment:
`rentedVehicle` is only set to true if the mode is `MICROMOBILITY_RENT`.

Reply:
Thanks for the clarification. Updated in https://github.com/ibi-group/otp-middleware/pull/91/commits/43425abecb53ee8d559c0d7c75062e6c84b7f03d.